File: 5Semestre/Disp Moveis II/Desafios/01-AppMegaSena/app/src/main/java/br/gov/br/cps/megasena/MainActivity.java
package br.gov.br.cps.megasena;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.HashSet;
import java.util.Random;

public class MainActivity extends AppCompatActivity {

    TextView n1, n2, n3, n4, n5, n6;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        n1 = findViewById(R.id.n1);
        n2 = findViewById(R.id.n2);
        n3 = findViewById(R.id.n3);
        n4 = findViewById(R.id.n4);
        n5 = findViewById(R.id.n5);
        n6 = findViewById(R.id.n6);
    }

    public void sorteioNumero(View view) {
        Random random = new Random();
        HashSet<Integer> numeros = new HashSet<>();

        while (numeros.size() < 6) {
            numeros.add(random.nextInt(60) + 1);
        }

        Integer[] lista = numeros.toArray(new Integer[0]);

        n1.setText(String.valueOf(lista[0]));
        n2.setText(String.valueOf(lista[1]));
        n3.setText(String.valueOf(lista[2]));
        n4.setText(String.valueOf(lista[3]));
        n5.setText(String.valueOf(lista[4]));
        n6.setText(String.valueOf(lista[5]));
    }

    public void limparNumeros(View view){
        n1.setText("");
        n2.setText("");
        n3.setText("");
        n4.setText("");
        n5.setText("");
        n6.setText("");
    }

}

File: 4Semestre/Disp. Moveis I/Android Studio/Ex 102/app/src/main/java/com/example/calcbitcoin/MainActivity.kt
package com.example.calcbitcoin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.*
import java.net.URL
import org.json.JSONObject
import java.text.NumberFormat
import java.util.*
import kotlinx.coroutines.*
import kotlinx.coroutines.Dispatchers.IO
import kotlinx.coroutines.Dispatchers.Main
import androidx.appcompat.app.AppCompatDelegate

class MainActivity : AppCompatActivity() {

    val API_URL = "https://www.mercadobitcoin.net/api/BTC/ticker/"
    var cotacaoBitcoin: Double = 0.0

    private lateinit var btnCalcular: Button
    private lateinit var txtValor: EditText
    private lateinit var txtCotacao: TextView
    private lateinit var txtQtdBitcoins: TextView
    private lateinit var btnTema: Button
    private lateinit var layoutPrincipal: LinearLayout

    private var modoEscuroAtivo = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        txtValor = findViewById(R.id.txtValor)
        btnCalcular = findViewById(R.id.btnCalcular)
        txtCotacao = findViewById(R.id.txtCotacao)
        txtQtdBitcoins = findViewById(R.id.txtQtdBitcoins)
        btnTema = findViewById(R.id.btnTema)
        layoutPrincipal = findViewById(R.id.layoutPrincipal)

        buscarCotacao()

        btnCalcular.setOnClickListener { calcular() }

        btnTema.setOnClickListener {
            modoEscuroAtivo = !modoEscuroAtivo

            if (modoEscuroAtivo) {
                layoutPrincipal.setBackgroundColor(getColor(R.color.fundo_azul_escuro))
                txtCotacao.setTextColor(getColor(R.color.white))
                txtQtdBitcoins.setTextColor(getColor(R.color.white))
                btnTema.setTextColor(getColor(R.color.white))
                btnTema.text = "Modo Claro"
            } else {
                layoutPrincipal.setBackgroundColor(getColor(R.color.fundo_azul_claro))
                txtCotacao.setTextColor(getColor(R.color.black))
                txtQtdBitcoins.setTextColor(getColor(R.color.black))
                btnTema.setTextColor(getColor(R.color.black))
                btnTema.text = "Modo Escuro"
            }
        }
    }

    fun buscarCotacao() {
        GlobalScope.launch(IO) {
            try {
                val resposta = URL(API_URL).readText()
                cotacaoBitcoin = JSONObject(resposta).getJSONObject("ticker").getDouble("last")

                val f = NumberFormat.getCurrencyInstance(Locale("pt", "BR"))
                val cotacaoFormatada = f.format(cotacaoBitcoin)

                withContext(Main) {
                    txtCotacao.text = cotacaoFormatada
                }
            } catch (e: Exception) {
                e.printStackTrace()
                withContext(Main) {
                    Toast.makeText(this@MainActivity, "Erro ao buscar cotação.", Toast.LENGTH_LONG).show()
                }
            }
        }
    }

    fun calcular() {
        if (txtValor.text.isEmpty()) {
            txtValor.error = "Preencha um valor"
            return
        }

        val valorDigitado = txtValor.text.toString().replace(",", ".").toDouble()
        val resultado = if (cotacaoBitcoin > 0) valorDigitado / cotacaoBitcoin else 0.0
        txtQtdBitcoins.text = "%.8f".format(resultado)
    }
}
